src/main/java/org/gradle/profiler/jfr/JFRJvmArgsCalculator.java

```diff
@@ -29,16 +29,21 @@ public void calculateJvmArgs(List<String> jvmArgs) {
             jvmArgs.add("-XX:+UnlockCommercialFeatures");
         }
         jvmArgs.add("-XX:+FlightRecorder");
-        jvmArgs.add("-XX:FlightRecorderOptions=stackdepth=1024");
         jvmArgs.add("-XX:+UnlockDiagnosticVMOptions");
         jvmArgs.add("-XX:+DebugNonSafepoints");
+
+        StringBuilder flightRecorderOptions = new StringBuilder("-XX:FlightRecorderOptions=stackdepth=1024");
+
         if (profileOnStart) {
-            String startArgs = "name=profile,settings=" + args.getJfrSettings();
+            jvmArgs.add("-XX:StartFlightRecording=name=profile,settings=" + args.getJfrSettings());
             if (captureOnExit) {
-                startArgs += ",dumponexit=true,filename=" + outputFile.getAbsolutePath();
+                flightRecorderOptions.append(",defaultrecording=true,dumponexit=true")
+                    .append(",dumponexitpath=")
+                    .append(outputFile.getParentFile().getAbsolutePath());
             }
-            jvmArgs.add("-XX:StartFlightRecording=" + startArgs);
         }
+
+        jvmArgs.add(flightRecorderOptions.toString());
```

Comment:
According to https://docs.oracle.com/javacomponents/jmc-5-4/jfr-runtime-guide/run.htm#JFRUH164 it should be `-XX:FlightRecorderOptions=defaultrecording=true,dumponexit=true,dumponexitpath=path`. Previous `-XX:StartFlightRecording=dumponexit=true` is wrong.